src/page-components/collections/collection-page.astro

```diff
@@ -0,0 +1,142 @@
+---
+import { ExtendedCollectionInfo, ExtendedPostInfo } from "../../types";
+import { MarkdownInstance } from "astro";
+import { Picture } from "@astrojs/image/components";
+import { Picture as UUPicture } from "components/base/image/picture";
+import { LargeButton, Button } from "src/components/base/button/button";
+import { translate } from "src/utils/translations";
+
+import "../../styles/post-body.scss";
+import "../../styles/tabs.scss";
+import "../../styles/shiki.scss";
+import "../../styles/convertkit.scss";
+import { getImageSize } from "../../utils/get-image-size";
+
+import styles from "./collection-page.module.scss";
+import { collections } from "../../utils/data";
+import { getUnicornProfilePicMap } from "utils/get-unicorn-profile-pic-map";
+
+const unicornProfilePicMap = await getUnicornProfilePicMap();
+
+const posts = (
+  await Astro.glob<ExtendedPostInfo>("../../../content/blog/**/*.md")
+).map((post) => post.frontmatter);
```

Comment:
I'd like to use `getPostsByCollection(collection.slug)` from `utils/api.ts` instead of `Astro.glob` here, since it's only using the post frontmatter - it'll both be nicer for build perf, and make it easier to mess with the dir structure in the future.

Reply:
We can't - this isn't just posts related to the collection - it's all posts by the author